pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -71,6 +75,15 @@ type projectLifecycle struct {
 
 func (l *projectLifecycle) sync(key string, orig *v3.Project) (runtime.Object, error) {
 	if orig == nil || orig.DeletionTimestamp != nil {
+		projectID := ""
+		splits := strings.Split(key, "/")
+		if len(splits) == 2 {
+			projectID = splits[1]
```

Comment:
what should happen if projectID doesn't get assigned?

Reply:
I think it'll always get assigned since the key will be of the form clusterID/projectID.